scipy/spatial/transform/_rotation_xp.py

```diff
@@ -1061,7 +1080,7 @@ def _get_angles(
     case1 = xp.abs(angles[..., 1]) <= eps
     case2 = xp.abs(angles[..., 1] - xp.pi) <= eps
     case0 = ~(case1 | case2)
-    if not is_lazy_array(case0) and xp.any(~case0):
+    if not is_lazy_array(case0) and xp.any(~case0) and not suppress_warnings:
```

Comment:
we can take advantage of short-circuiting (https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not):

```suggestion
    if not suppress_warnings and not is_lazy_array(case0) and xp.any(~case0):
```

Reply:
My thinking was that `suppress_warnings` is false by default, so we'd be more likely to short-circuit with the gimbal lock check first. But I suppose anybody who cares about such a tiny speedup will probably set `suppress_warnings` to true anyway, so I think you're right about flipping the order.

Reply:
I think same as other `_get_angles`, this should not emit warnings. 
Emitting warnings is an expensive operation. So you want to return whether there is a gimbal lock, and emit the warning once at the call site if there was a gimbal lock.

Reply:
I may be confused, but I think this function is only called once: the new backend uses broadcasting instead of calling this function in a loop.